File: src/priv/bajdcc/LALR1/grammar/runtime/RuntimeInstNon.java
package priv.bajdcc.LALR1.grammar.runtime;

/**
 * 【中间代码】零元操作数指令
 *
 * @author bajdcc
 */
public class RuntimeInstNon extends RuntimeInstBase {

	public RuntimeInstNon(RuntimeInst inst) {
		this.inst = inst;
	}

	@Override
	public int getAdvanceLength() {
		return 1;
	}

	@Override
	public void gen(ICodegenByteWriter writer) {
		writer.genInst(inst);
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/grammar/runtime/RuntimeInstNon.kt
package com.bajdcc.LALR1.grammar.runtime

/**
 * 【中间代码】零元操作数指令
 *
 * @author bajdcc
 */
open class RuntimeInstNon(inst: RuntimeInst) : RuntimeInstBase() {

    override val advanceLength: Int
        get() = 1

    init {
        this.inst = inst
    }

    override fun gen(writer: ICodegenByteWriter) {
        writer.genInst(inst)
    }
}
